### PR TITLE
refactor(utils): hoist `utils::utils` into `utils`

### PR DESCRIPTION
**Pull request by rami3l**
Continuation of #3882.

### CODE REVIEW

**Comment by rami3l** [comment]
Sorry for continuous force-pushing... This was mostly related to formatting issues after global regex-based replacements ('~')

Anyway, it should be alright now.

File: src/utils/mod.rs [hunk 1]
```diff
@@ -1,9 +1,741 @@
 //!  Utility functions for Rustup
 pub(crate) mod notifications;
+pub(crate) mod notify;
```
Comment (djc) [inline src/utils/mod.rs]: I think we should put the module declarations below the imports.

File: src/utils/mod.rs [hunk 2]
```diff
@@ -1,9 +1,741 @@
...
+use retry::{retry, OperationResult};
+use sha2::Sha256;
+use url::Url;
+
+use crate::errors::*;
+use crate::process::Process;
+
+#[cfg(not(windows))]
+pub(crate) use crate::utils::raw::find_cmd;
+pub(crate) use crate::utils::raw::is_directory;
```
Comment (djc) [inline src/utils/mod.rs]: Let's group all the `crate` uses into a single block?

File: src/cli/self_update/unix.rs [hunk 3]
```diff
@@ -7,7 +7,7 @@ use tracing::{error, warn};
 use super::install_bins;
 use super::shell;
 use crate::process::Process;
-use crate::utils::utils;
+use crate::utils;
 use crate::utils::Notification;
```
Comment (djc) [inline src/cli/self_update/unix.rs]: Group this into `{self, Notification}`?

File: src/cli/self_update/windows.rs [hunk 4]
```diff
@@ -23,7 +23,7 @@ use super::{install_bins, report_error, InstallOpts};
 use crate::cli::{download_tracker::DownloadTracker, markdown::md};
 use crate::dist::TargetTriple;
 use crate::process::{terminalsource::ColorableTerminal, Process};
-use crate::utils::utils;
+use crate::utils;
 use crate::utils::Notification;
```
Comment (djc) [inline src/cli/self_update/windows.rs]: And here.

File: src/dist/component/package.rs [hunk 5]
```diff
@@ -18,8 +18,8 @@ use crate::dist::component::transaction::*;
 use crate::dist::temp;
 use crate::errors::*;
 use crate::process::Process;
+use crate::utils;
```
Comment (djc) [inline src/dist/component/package.rs]: And here (can simplify to just `Notification`).

File: src/dist/component/tests.rs [hunk 6]
```diff
@@ -9,8 +9,8 @@ use crate::dist::Notification;
 use crate::dist::DEFAULT_DIST_SERVER;
 use crate::errors::RustupError;
 use crate::process::TestProcess;
+use crate::utils;
```
Comment (djc) [inline src/dist/component/tests.rs]: `{utils, raw as utils_raw}`.

File: src/dist/temp.rs [hunk 7]
```diff
@@ -7,9 +7,9 @@ use std::path::{Path, PathBuf};
 pub(crate) use anyhow::{Context as _, Result};
 use thiserror::Error as ThisError;
 
+use crate::utils;
```
Comment (djc) [inline src/dist/temp.rs]: `{self, raw}`.

**Review by djc — APPROVED** [review]
Nice!

File: src/utils/mod.rs (hunk 1 above)
Comment (rami3l) [reply src/utils/mod.rs]: @djc Can I do that in a separate commit? I was trying to minimize the LoDiff back then...

Comment (djc) [reply src/utils/mod.rs]: Sure.

**Review by djc — APPROVED** [review]

**Pull request merged**